### PR TITLE
Feat 1657 - invoice / financial report related database models implementation

### PR DESCRIPTION
**Pull request by phpgeekfr**
Feat 1657 - invoice / financial report related database models implementation

### CODE REVIEW

File: libs/utils/src/lib/common-interfaces/terms.ts [hunk 1]
```diff
@@ -54,12 +55,22 @@ export interface ScheduledDateRaw<D> extends TermsRaw<D> {
   period: FloatingDuration, // @dev this replace floatingDuration for better readability
 }
 
+/**
+ * A ScheduledDateRaw interface with a counter to keep schedule order
+ */
+export interface ScheduledDateWithCounterRaw<D> extends ScheduledDateRaw<D> {
+  count: 0;
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/common-interfaces/terms.ts]: count has to be "always" 0 ??

File: libs/utils/src/lib/common-interfaces/terms.ts [hunk 2]
```diff
@@ -79,3 +90,11 @@ export function createScheduledDate(params: Partial<ScheduledDate> = {}): Schedu
     period: createFloatingDuration(params ? params.period : undefined),
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/common-interfaces/terms.ts]: Why not doing `createFloatingDuration(params.period)` directly ? I don't understand the difference.

File: libs/financial-report/src/lib/version/+state/financial-report-version.model.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+/** An interface for a single document to display versions subcollection count. */
+export interface VersionMeta extends FinancialReportVersion {
+  count?: number;
+}
+
+export function createVersionMeta(params: Partial<VersionMeta>): VersionMeta {
+  return {
+    id: '_meta',
+    count: params && params.count || 0,
+  } as VersionMeta;
```
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/financial-report/src/lib/version/+state/financial-report-version.model.ts]: This is intended as VersionMeta will always contains only `count`

File: libs/utils/src/lib/common-interfaces/terms.ts (hunk 2 above)
Comment (phpgeekfr) [reply libs/utils/src/lib/common-interfaces/terms.ts]: no reason ..

**Pull request merged**